### PR TITLE
feat(policy-gen): add flags to generate 'targetRef', 'to' and 'from

### PR DESCRIPTION
**Pull request by lobkovilya**
Add 3 new flags for policy-gen:
- generateTargetRef
- generateTo
- genrateFrom

These flags allow generating proto and validation.

### Checklist prior to review

- [X] Link to docs PR or issue -- N/A
- [X] Link to UI issue or PR -- N/A
- [X] Is the [issue worked on linked][1]? -- N/A
- [X] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [X] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Unit Tests --
- [ ] E2E Tests --
- [ ] Manual Universal Tests --
- [ ] Manual Kubernetes Tests --
- [X] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? -- No
- [X] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? -- No
- [X] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch)? No

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/plugins/policies/donothingpolicy/api/v1alpha1/validator.go [hunk 1]
```diff
@@ -1,11 +1,65 @@
...
+
+			defaultField := to.Index(idx).Field("default")
+			if toItem.GetDefault() == nil {
+				verr.AddViolationAt(defaultField, "cannot be nil")
+			} else {
+				// TODO add default conf validation
+				verr.AddViolationAt(defaultField, "")
+			}
+		}
+	}
```
Comment (lahabana) [inline pkg/plugins/policies/donothingpolicy/api/v1alpha1/validator.go:62]: I guess typing makes things harder to move this to a generic method in the `matchers` package?

File: tools/policy-gen/bootstrap/root.go [hunk 2]
```diff
@@ -157,6 +167,7 @@ package kuma.plugins.policies.{{ .nameLower }}.{{ .version }};
 import "mesh/options.proto";
 option go_package = "{{ .module }}/{{.nameLower}}/api/{{ .version }}";
 
+import "common/v1alpha1/targetref.proto";
```
Comment (lahabana) [inline tools/policy-gen/bootstrap/root.go:170]: If you want to do it right you need to add a `{{- if .generateTargetRef || .generate.To || .generateFrom}}`

File: tools/policy-gen/bootstrap/root.go [hunk 3]
```diff
@@ -214,13 +266,77 @@ func (p plugin) Apply(rs *core_xds.ResourceSet, ctx xds_context.Context, proxy *
 
 var validatorTemplate = template.Must(template.New("").Option("missingkey=error").Parse(
 	`package {{.version}}
+
 import (
+	common_proto "github.com/kumahq/kuma/api/common/v1alpha1"
 	"github.com/kumahq/kuma/pkg/core/validators"
+	matcher_validators "github.com/kumahq/kuma/pkg/plugins/policies/matchers/validators"
```
Comment (lahabana) [inline tools/policy-gen/bootstrap/root.go:273]: Same for here

**Review by lahabana — APPROVED** [review]
Only nits! But this is very cool! This will save so much time to future implems of policies.

File: pkg/plugins/policies/donothingpolicy/api/v1alpha1/validator.go (hunk 1 above)
Comment (lobkovilya) [reply pkg/plugins/policies/donothingpolicy/api/v1alpha1/validator.go:62]: Exactly, `From` and `To` always have different types depending on the policy

**Pull request merged**